search/v2/geocoder.cpp

```diff
@@ -276,14 +281,25 @@ void Geocoder::Go(vector<FeatureID> & results)
 
       m_matcher.reset(new FeaturesLayerMatcher(m_index, *m_context, *this /* cancellable */));
 
+      unique_ptr<coding::CompressedBitVector> viewportCBV;
+      if (inViewport)
+      {
+        viewportCBV = Retrieval::RetrieveGeometryFeatures(
+            m_context->m_value, static_cast<my::Cancellable const &>(*this), m_params.m_viewport, m_params.m_scale);
+      }
+
       // Creates a cache of posting lists for each token.
       m_addressFeatures.resize(m_numTokens);
       for (size_t i = 0; i < m_numTokens; ++i)
       {
         PrepareRetrievalParams(i, i + 1);
+
         m_addressFeatures[i] = Retrieval::RetrieveAddressFeatures(
             m_context->m_value, *this /* cancellable */, m_retrievalParams);
         ASSERT(m_addressFeatures[i], ());
+
+        if (viewportCBV)
```

Comment:
In MatchViewportAndPosition we again retrieve features around viewport and position. Moreover, in map search we do not need to search around position.